aws/infrastructure.tf

```diff
@@ -95,6 +133,13 @@ resource "aws_instance" "instances" {
   }
 
   depends_on = [aws_internet_gateway.gw]
+
+  # spot specific variables
+  wait_for_fulfillment   = lookup(each.value, "wait_for_fulfillment", true)
+  spot_type              = lookup(each.value, "spot_type", "one-time")
+  spot_price             = lookup(each.value, "spot_price", null)
+  block_duration_minutes = lookup(each.value, "block_duration_minutes", null)
```

Comment:
How about a minimum duration so that you know it is worth your while to spin up the node (given it will take some time to configure)

Reply:
I tried to set `block_duration_minutes` to 60, but I encountered a weird `Max Spot Instances Count` error that dissappeared when I removed the attribute. This requires more testing.